src/charm.py

```diff
@@ -940,569 +472,497 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
+            try:
+                client.delete_mfa_credential(
+                    identity_id, mfa_type="oidc", params={"identifier": credential_id}
+                )
+            except IdentityCredentialsNotExistError:
+                event.log(f"Identity {identity_id} has no oidc credentials")
+                return
+            except ClientRequestError:
+                event.fail("Failed to delete the oidc credentials")
                 return
-        except requests.exceptions.RequestException as e:
-            event.fail(f"Failed to request Kratos API: {e}")
-            return
 
-        event.log("Account has been unlinked successfully")
+        event.log(f"Successfully unlink the oidc account for identity {identity_id}")
 
     def _on_invalidate_identity_sessions_action(self, event: ActionEvent) -> None:
-        if not self._kratos_service_is_running:
+        if not self._workload_service.is_running:
             event.fail("Service is not ready. Please re-run the action when the charm is active")
             return
 
-        if not (identity_id := self._get_identity_id(event)):
-            return
-
-        event.log("Invalidating user sessions")
-        try:
-            sessions_invalidated = self.kratos.invalidate_sessions(identity_id=identity_id)
-            if not sessions_invalidated:
-                event.log("User has no sessions")
+        identity_id = event.params.get("identity-id")
+        with HTTPClient(base_url=f"http://127.0.0.1:{KRATOS_ADMIN_PORT}") as client:
+            try:
+                client.invalidate_sessions(identity_id)
+            except IdentitySessionsNotExistError:
+                event.log(f"Identity {identity_id} has no sessions")
+                return
+            except ClientRequestError:
+                event.fail(f"Failed to invalidate and delete sessions for identity {identity_id}")
                 return
-        except requests.exceptions.RequestException as e:
-            event.fail(f"Failed to request Kratos API: {e}")
-            return
 
-        event.log("User sessions have been invalidated and deleted")
+        event.log("Successfully invalidated and deleted the sessions")
 
     def _on_reset_identity_mfa_action(self, event: ActionEvent) -> None:
-        if not self._kratos_service_is_running:
+        if not self._workload_service.is_running:
             event.fail("Service is not ready. Please re-run the action when the charm is active")
             return
 
+        identity_id = event.params.get("identity-id")
         mfa_type = event.params.get("mfa-type")
-        if not mfa_type:
-            event.fail("MFA type must be specified")
-            return
-
-        if mfa_type not in ("totp", "lookup_secret", "webauthn"):
-            event.fail(
-                f"Unsupported MFA credential type {mfa_type}, "
-                "allowed methods are: `totp`, `lookup_secret` and `webauthn`"
-            )
-            return
```

Comment:
I think we should still check the mfa type e.g. to catch input mistakes like `lookupsecret`